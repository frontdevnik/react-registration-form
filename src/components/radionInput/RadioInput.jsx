import React from 'react';

export default ({ name, text = '', register, checked }) => {
  return (
    <label>
      <input
        name={name}
        type="radio"
        ref={register()}
        checked={checked}
        value={text}
      />
      <span>{text}</span>
    </label>
  );
};
