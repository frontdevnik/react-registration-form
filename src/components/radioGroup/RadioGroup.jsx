import React from 'react';

import RadioInput from '../radionInput';

export default ({ values, name, register }) =>
  values.map((value) => (
    <RadioInput name={name} register={register} text={value} key={value} />
  ));
