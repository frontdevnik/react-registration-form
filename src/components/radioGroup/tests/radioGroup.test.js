import React from 'react';

import RadioInput from "../../radionInput";
import RadioGroup from '../RadioGroup';

import { radioGroupsProps } from './stubProps';

describe('<RadioGroup />', () => {
  const component = mount(<RadioGroup {...radioGroupsProps} />);

  it('should render correctly', () => {
    expect(component.html()).toMatchSnapshot();
  });
  
  it('should elements display correctly', () => {
    expect(component.find(RadioInput).length).toBe(radioGroupsProps.values.length);
  });
});
