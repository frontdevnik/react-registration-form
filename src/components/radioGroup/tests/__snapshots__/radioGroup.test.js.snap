// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should render correctly 1`] = `"<label><input name=\\"values\\" type=\\"radio\\" value=\\"value1\\"><span>value1</span></label><label><input name=\\"values\\" type=\\"radio\\" value=\\"value2\\"><span>value2</span></label>"`;
